src/config/findHomeDir.ts

```diff
@@ -0,0 +1,76 @@
+import fs from 'fs';
+import path from 'path';
+
+// Implement the same logic as in
+// https://github.com/kubernetes/kubernetes/blob/ea07644/staging/ \
+// src/k8s.io/client-go/util/homedir/homedir.go
+
+function dropDuplicatesAndNils(a: Array<string | null | undefined>): string[] {
+  return a.reduce((acceptedValues, currentValue) => {
+    // Good-enough algorithm for reducing a small (3 items at this point) array into an ordered list
+    // of unique non-empty strings.
+    if (currentValue && !acceptedValues.includes(currentValue)) {
+      return acceptedValues.concat(currentValue);
+    } else {
+      return acceptedValues;
+    }
+  }, [] as string[]);
+}
+
+export function findHomeDir(): string | null {
+  if (process.platform !== 'win32') {
+    if (process.env.HOME) {
+      try {
+        fs.accessSync(process.env.HOME);
+
+        return process.env.HOME;
+      } catch (ignore) { }
+    }
+
+    return null;
+  }
+
+  // $HOME is always favored, but the k8s go-client prefers the other two env vars
```

Comment:
```suggestion
  // $HOME is always favoured, but the k8s go-client prefers the other two env vars
```
(Because your variable names contain a `u`)